db/change_cache_test.go

```diff
@@ -226,6 +226,126 @@ func TestLateSequenceHandlingWithMultipleListeners(t *testing.T) {
 
 }
 
+// Verify that a continuous changes feed hitting an error when building its late sequence feed will roll back to
+// its low sequence value, then recover and successfully send subsequent late sequences.
+func TestLateSequenceErrorRecovery(t *testing.T) {
+
+	if base.TestUseXattrs() {
+		t.Skip("This test does not work with XATTRs due to calling WriteDirect().  Skipping.")
+	}
+
+	defer base.SetUpTestLogging(base.LevelInfo, base.KeyChanges|base.KeyCache)()
+
+	db, testBucket := setupTestDBWithCacheOptions(t, shortWaitCache())
+	defer tearDownTestDB(t, db)
+	defer testBucket.Close()
+
+	db.ChannelMapper = channels.NewDefaultChannelMapper()
+
+	// Create a user with access to channel ABC
+	authenticator := db.Authenticator()
+	assert.True(t, authenticator != nil, "db.Authenticator() returned nil")
+	user, err := authenticator.NewUser("naomi", "letmein", channels.SetOf(t, "ABC"))
+	assert.NoError(t, err, fmt.Sprintf("Error creating new user: %v", err))
+	authenticator.Save(user)
+
+	// Start continuous changes feed
+	var options ChangesOptions
+	options.Since = SequenceID{Seq: 0}
+	options.Terminator = make(chan bool)
+	defer close(options.Terminator)
+	options.Continuous = true
+	options.Wait = true
+	feed, err := db.MultiChangesFeed(base.SetOf("ABC"), options)
+	goassert.True(t, err == nil)
+
+	// Reads events until it gets a nil event, which indicates the changes loop has entered wait mode.
+	// Returns the last non-nil event it
+	nextFeedIteration := func() []*ChangeEntry {
+		events := make([]*ChangeEntry, 0)
+		for {
+			select {
+			case event := <-feed:
+				if event == nil {
+					return events
+				}
+				events = append(events, event)
+			case <-time.After(10 * time.Second):
+				assert.Fail(t, "Expected sequence didn't arrive over feed")
+				return nil
+			}
+		}
+	}
+
+	nextEvents := nextFeedIteration()
+	assert.Equal(t, len(nextEvents), 0) // Empty feed indicates changes is in wait mode
+
+	// Write sequence 1, wait for it on feed
+	WriteDirect(db, []string{"ABC"}, 1)
+
+	nextEvents = nextFeedIteration()
+	assert.Equal(t, len(nextEvents), 1)
```

Comment:
Swapped to require to prevent nil pointer below

```suggestion
	require.Equal(t, len(nextEvents), 1)
```